streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -2927,7 +2927,30 @@ void runOnce() {
 
     @Test
     public void shouldCheckStateUpdater() {
+        final StreamThread streamThread = setUpThreadWithStateUpdater(StreamsTestUtils.getStreamsConfig());
+        final TaskManager taskManager = streamThread.taskManager();
+        streamThread.setState(State.STARTING);
+
+        streamThread.runOnce();
+
+        Mockito.verify(taskManager).checkStateUpdater(Mockito.anyLong(), Mockito.any());
+        Mockito.verify(taskManager).process(Mockito.anyInt(), Mockito.any());
+    }
+
+    @Test
+    public void shouldRespectPollTimeInPartitionsAssignedStateWithStateUpdater() {
```

Comment:
Could you also add tests for the other branches of the `if`-statement in `pollPhase()`?
I propose to add the case `state == State.PARTITIONS_ASSIGNED && !stateUpdaterEnabled` in this PR and to add the other cases in a separate PR. 
With other cases I mean:
- `state == State.PARTITIONS_REVOKED`
- `state == State.RUNNING`
- `state == State.STARTING`
- `state == State.PENDING_SHUTDOWN`

Reply:
Yes, I will do this.